sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -128,6 +130,46 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
         ShimLoader.setSparkShimProviderClass(conf.shimsProviderOverride.get)
       }
 
+      // Compare if the cudf version mentioned in the classpath is equal to the version which plugin
+      // expects. If there is a version mismatch, throw error. This check can be disabled by
+      // setting this config spark.rapids.cudfVersionOverride=true
+      if (!conf.cudfVersionOverride) {
```

Comment:
Rather than say nothing on startup if the override config is set, I would like it to log a warning when the version found does not match what is expected (or if it cannot find the version).  Essentially the override property should still complain when the version doesn't match (or cannot be determined) but it's not fatal to startup.